interface/src/app/gene/genes.js

```diff
@@ -0,0 +1,376 @@
+}])
+
+// Directive for table containing search results
+.directive('searchResultTable', ['SearchResults', function( SearchResults ) {
+  return {
+    controller: ['$scope', 'SearchResults', function($scope, SearchResults) {
+      $scope.currentPage = 1;
+      $scope.itemsPerPage = 10;
+      $scope.totalResults = 0;
+      $scope.maxSize = 10;
+      $scope.resultsForPage = [];
+      $scope.searchResults = SearchResults.getSearchResults();
+      $scope.loadingSearchResults = false;
+
+      $scope.addAllNonAmbiguous = function() {
+      // Function to automatically add all genes that only have one search result
+        var searchResults = SearchResults.getSearchResults();
+        for (var key in searchResults) {
+          var results = searchResults[key];
+          if (results['found'].length <= 1) {
+            if (results['found'][0]) {
+              var gene = results['found'][0];
+              SearchResults.remove(key);
+            }
+          }
+        }
+      };
+
+      $scope.removeNotFound = function() {
+      // Function to get rid of all the queries that returned no results.
+          var searchResults = SearchResults.getSearchResults();
+          for (var key in searchResults) {
+            var results = searchResults[key];
+            if (results['found'].length === 0) {
+              SearchResults.remove(key);
+            }
+          }
+      };
+    }],
+
+    link: function(scope, element, attr) {
+      scope.$on('results.update', function() {
+        scope.totalResults = SearchResults.size();
+        var begin = ((scope.currentPage-1)*scope.itemsPerPage);
+        var end = begin + scope.itemsPerPage;
+        scope.resultsForPage = SearchResults.getQueries().slice(begin, end);
+      });
+
+      scope.$on('results.loadingSearchResults', function() {
+        scope.loadingSearchResults = true;
+      });
+
+      scope.$on('results.searchResultsReturned', function() {
+        scope.loadingSearchResults = false;
+      });
+
+      //Watch for page changes and update
+      scope.$watch('currentPage', function() {
+        var begin = ((scope.currentPage-1)*scope.itemsPerPage);
+        var end = begin + scope.itemsPerPage;
+        scope.resultsForPage = SearchResults.getQueries().slice(begin, end);
+      });
+    },
+    replace: true,
+    restrict: "E",
+    scope: true,
+    templateUrl: 'gene/search-result-table.tpl.html'
+    };
+}])
+
+// Directive for table containing search results
+.directive('selectedGenesPanel', function( ) {
+    return {
+        controller: ['$scope', 'SelectedGenesFactory', '$state',
+          function($scope, SelectedGenesFactory, $state) {
+            $scope.selectedGenes = SelectedGenesFactory.return_genes();
+
+            $scope.sendToNetwork = function () {
+              var genes = [];
+              for (var gene in $scope.selectedGenes) {
+                genes.push(gene);
+              }
+              $state.go('gene_network', {'genes': genes});
```

Comment:
Yes, I forgot that **scope.selectedGenes** is actually an object. In my comment on line 85, I was actually asking whether it is possible to change **selected_genes** from object to an array.